src/operator/tensor/control_flow_op.h

```diff
@@ -188,7 +188,7 @@ inline bool WhereOpShape(const nnvm::NodeAttrs& attrs,
     SHAPE_ASSIGN_CHECK(*in_attrs, 0, tshape);
     return true;
   } else if ((*in_attrs)[0].ndim() == 1) {
-    return (*in_attrs)[0].Size() == static_cast<size_t>(tshape[0]);
+    CHECK_EQ((*in_attrs)[0].Size(), static_cast<size_t>(tshape[0]));
```

Comment:
this is wrong. what is the problem of the original code? If the first dimension doesn't match, infer shape fails, so it should return false.